libtextsecure/sendmessage.js

```diff
@@ -379,8 +379,14 @@ MessageSender.prototype = {
     const syncMessages = await Promise.all(
       chunked.map(c => libloki.api.createContactSyncMessage(c))
     );
+    const pubKey = textsecure.storage.user.getNumber();
+
+    const currentPubKey = new libsession.Types.PubKey(pubKey);
+
     const syncPromises = syncMessages.map(syncMessage =>
-      libsession.getMessageQueue().sendSyncMessage(syncMessage)
+      libsession
```

Comment:
The previous change was fine as you're sending a proper sync message.
```
libsession.getMessageQueue().sendSyncMessage(syncMessage)
```

I think you got confused with what we were doing in `conversation.js`. The reason we didn't call `sendSyncMessage` there for stuff like expiration update is that the expiration message was being sent to the user (doesn't matter if it's the current user or not) anyway in the lines below.

Here we want to send a sync message to ourselves and not any other user.